### PR TITLE
Remove unnecessary randomness of mole

### PR DESCRIPTION
**Pull request by weluvgoatz**
Having some enemies with random behavior is so horrible for stuff like speedruns, or situations in which something should not be random.  There is an instance of randomness in badguys that literally does not need to exist, and I have removed it here.  That instance is this:
- Random throwing of rocks by moles has been changed to a certain pattern, which is still very similar to the old one.

### CODE REVIEW

**Comment by Rusty-Box** [comment]
Zeekling's random speed is already addressed in #2624

**Comment by Alasdairbugs** [comment]
i believe skullyhops also have some random function which affect their delay before hopping

**Review by Frostwithasideofsalt — APPROVED** [review]

File: src/badguy/mole.cpp [hunk 1]
```diff
@@ -75,14 +76,13 @@ Mole::collision_squished(GameObject& )
 void
 Mole::throw_rock()
 {
-  float angle;
-  float base_angle = (m_flip == NO_FLIP ? 90.0f : 270.0f);
-  angle = math::radians(gameRandom.randf(base_angle - 15.0f, base_angle + 15.0f));
-  float vx = cosf(angle) * THROW_VELOCITY;
-  float vy = -sinf(angle) * THROW_VELOCITY;
+  float angle = math::radians(135.f - (float(cycle_num) * 30.f));
```
Comment (Vankata453) [inline src/badguy/mole.cpp:79]: Use `static_cast` instead:

```suggestion
  float angle = math::radians(135.f - (static_cast<float>(cycle_num) * 30.f));
```

**Review by Vankata453 — APPROVED** [review]
Works well! The consistent pattern also allows making strategies on how to go around a mole in certain cases possible.

Code-wise, one little thing:

File: src/badguy/mole.cpp (hunk 1 above)
Comment (MatusGuy) [reply src/badguy/mole.cpp:79]: Look! It's a giant commit button

Comment (tobbi) [reply src/badguy/mole.cpp]: I want to be able to squash unrelated commit suggestions.

**Pull request merged**